### DISCUSSION TITLE
Version upgrade changing random forest VIMP algorithm

### DISCUSSION THREAD

**Discussion by llangset** [kogalur/randomForestSRC#352] [Q&A]
I have a question regarding some coding and details of the package RandomForestSRC. I had been working with both an older version of R and an older version of your package. I updated both (due to another project) and am now getting different results. Some of my VIMP results have changed quite substantially. Were the defaults for VIMP changed? Or do I need to look elsewhere? The function I am using is gg_mininal_depth which gives both the minimal depth and VIMP. Note: MD results are slightly different too, but less obviously so.

**Comment by ishwaran** [comment]
Yes the default choice of VIMP has changed starting from a few versions ago from permutation VIMP to so-called "anti VIMP".  From the help file for `rfsrc`

```
New items to be aware of:

       1. For computational speed, the default VIMP is no longer
          "permute" (Breiman-Cutler permutation importance) and has
          been switched to "anti" (‘importance="anti"’,
          ‘importance=TRUE’; see below for details).  Be aware in some
          situations, such as highly imbalanced classification, that
          permutation VIMP may perform better.  Permutation VIMP is
          obtained using ‘importance="permute"’.
```

Therefore, in order to get results similar to before, you have to manually request permutation VIMP via `importance="permute"`

**Reply by llangset** [reply]
So besides speed what are the merits, demerits of permutation VIMP (permute), random daughter VIMP (random), and anti? Are there any performance benchmarks? Do you have references?

**Comment by ishwaran** [comment]
The merit of anti is it's faster.  Probably permutation is the best of the VIMP procedures.  But I think all of these methods will be replaced soon with a new method I'm working on called VarPro.

**Comment by llangset** [comment]
Sounds interesting...I look forward to seeing it.

Lisa

**Comment by ishwaran** [comment]
There's a GitHub repos with preliminary software for trying it out

https://rdrr.io/github/kogalur/varPro/
